packages/vue/src/dotlottie.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2023 Design Barn Inc.
+ */
+
+import { type Config, DotLottie, type EventType, type EventListener, type Mode } from '@lottiefiles/dotlottie-web';
+import {
+  type VNode,
+  h,
+  ref,
+  onMounted,
+  type Ref,
+  watch,
+  type SetupContext,
+  onBeforeUnmount,
+  defineComponent,
+  toRefs,
+} from 'vue';
+
+export { type DotLottie };
+
+interface DotLottieVueProps extends Omit<Config, 'canvas'> {}
+
+export const DotLottieVue = defineComponent({
+  props: {
+    autoplay: { type: Boolean, required: false },
+    backgroundColor: { type: String, required: false },
+    data: { type: String, required: false },
+    loop: { type: Boolean, required: false },
+    mode: { type: String as () => Mode, required: false },
+    renderConfig: { type: Object, required: false },
+    segments: { type: Array as unknown as () => [number, number], required: false },
+    speed: { type: Number, required: false },
+    src: { type: String, required: false },
+    useFrameInterpolation: { type: Boolean, required: false },
+  },
```

Comment:
can we infer the `type` of each prop from the dotlottie-web `Config` type. i wonder if we can just pass the type of the `DotLottieVueProps` created earlier 👀

Reply:
these are not typescript types. its vue types feature, the first param of setup function takes the  `setup(props: DotLottieVueProps,..)` typescript types. Looks like both needs to be there to work. Also can't compile if both doesn't match haha